### PR TITLE
add bond calibration test

### PR DESCRIPTION
**Pull request by noittom**
adding a test with bond calibration of a Nelson-Siegel-Svensson curve to a given set of bond.
currently I think I have some problems with using finmath lib as interpolated yield values are quite different from actual yields.

I would appreciate it if you could have a look.

The main idea is that Solver should be used with some current bond prices to calibrate NSS curve

Alexey Monakhov

### CODE REVIEW

**Comment by noittom** [comment]
any updates ?

**Comment by cfries** [comment]
Sorry for the delay! Will have a look at it now!

**Comment by cfries** [comment]
There were two small issues:

- Your list of target prices was normed at the notion 100.0 (so having prices 99.0). However, the Bond evaluation did use a notion of 1.0.
- The product implementation Bond required a separate survival probability and basis curve. These two curves should have been dummy curves being identical to 1.0 - because you just like to have a single Bond curve as a discount curve (the three curves allow for a splitting into a risk free discount curve, a basis curve (e.g. industry) and an issuer specific part. I modified the product such that it can be used without this splitting.

Everything is our fault, because the documentation should have been better ;-)

File: src/test/java/net/finmath/marketdata/model/curves/NelsonSiegelSvenssonBondCalibrationTest.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	 * @throws SolverException Thrown if the solver cannot find a solution.
+	 */
+	@Test
+	public void testBondNSSCurveCalibration() throws SolverException {
+		//maturities in years.
+		double[] maturity = new double[]{0.1, 0.3, 1.0, 2.0, 3.0, 4.0, 5.0, 7.0, 10.0, 20.0, 30.0};
+		//coupon rates of bonds, 1.5% for all of them
+		double[] rates = new double[]{0.015, 0.015, 0.015, 0.015, 0.015, 0.015, 0.015, 0.015, 0.015, 0.015, 0.015};
+		//current prices of bonds, current price for face value of 100. (i.e. 99 means that the bond is sold with a discount)
+		double[] prices = new double[]{99.0, 99.8, 101.0, 101.5, 102.0, 102.8, 103.0, 103.8, 103.9, 104.0, 104.2};
```
Comment (cfries) [inline src/test/java/net/finmath/marketdata/model/curves/NelsonSiegelSvenssonBondCalibrationTest.java:47]: The bond assumes a notion of 1.0. So prices should be 0.99, 0.998, 1.01,...

Comment (cfries) [reply src/test/java/net/finmath/marketdata/model/curves/NelsonSiegelSvenssonBondCalibrationTest.java:47]: I have fixed this below (in a stream.map(x -> x/100)), but maybe it should be fixed here

**Review by cfries — APPROVED** [review]

**Comment by cfries** [comment]
I am merging this and will make some updates in the master branch. Maybe you can have a look there!

**Comment by cfries** [comment]
Your unit test is not on the master branch. I have added an assertion, checking for the price differences. Since the NSS curve has only a few parameters, there are some deviations, but overall it looks OK. Please have a look at your unit test in the master branch. Thanks for the PR and feedback.

I have removed the "Example Curve" you had added, because it looked worse to me.

Please get back to me if you have questions. I try to be quicker this time.

**Pull request merged**